entry.sh

```diff
@@ -33,7 +33,7 @@ fi
 # NOTE: this won't be necessary once the supervisor can update
 # itself, as using the label io.balena.features.journal-logs will
 # achieve the same objective
-if { [ ! -d /run/log/journal ] || [ -L /run/log/journal ]; } && [ -s "${ROOT_MOUNTPOINT}"/etc/machine-id ]; then
+if { [ ! -d /run/log/journal ] || [ -L /run/log/journal ]; } && [ -s "${STATE_MOUNTPOINT}"/machine-id ]; then
```

Comment:
Is this always the case? Even after HUP? Maybe we can check with the OS (if you have not done it yet)

Reply:
I tested this (see checklist below) and even on old devices, before and after HUP, the machine-id is present in the state partition at this path.